tests/Fixtures/TestIncentiveFinder.php

```diff
@@ -9,7 +9,25 @@
 
 class TestIncentiveFinder implements IncentiveFinder {
 
+	/**
+	 * @var Incentive[]
+	 */
+	private array $incentives;
+
+	/**
+	 * @param Incentive[] $incentives
+	 */
+	public function __construct( array $incentives ) {
+		$this->incentives = $incentives;
+	}
+
 	public function findIncentiveByName( string $name ): ?Incentive {
-		return new Incentive( $name );
+		$incentives = array_filter( $this->incentives, fn( $incentive ) => $incentive->getName() === $name );
+
+		if ( count( $incentives ) === 0 ) {
+			return null;
+		}
+
+		return reset( $incentives );
```

Comment:
Interesting, I'd have gone with `$incentives[0]`. But as long as PHPStan/PHPCS/Scrutinizer are not complaining about the potential `false` value (because I would have thought they aren't intelligent enough to recognize the count check), I'm fine with `reset`.

Reply:
I had that originally but `array_filter` seems to return an array with the keys from the original so if it was the second item it had the key `1`